divi/src/BlockFactory.cpp

```diff
@@ -65,28 +66,31 @@ bool BlockFactory::AppendProofOfStakeToBlock(
     CBlockTemplate& pBlockTemplate)
 {
     CBlock& block = pBlockTemplate.block;
-    CMutableTransaction txCoinStake;
     SetRequiredWork(pBlockTemplate);
     SetBlockTime(block);
 
     // ppcoin: if coinstake available add coinstake tx
     static int64_t nLastCoinStakeSearchTime = GetAdjustedTime(); // only initialized at startup
 
-    unsigned int nTxNewTime = 0;
-    if(coinstakeCreator_->CreateProofOfStake(
-            block.nBits,
-            block.nTime,
-            nLastCoinStakeSearchTime,
-            txCoinStake,
-            nTxNewTime))
-    {
+    CMutableTransaction txCoinStake;
+    if (customCoinstake_ != nullptr) {
+        txCoinStake = *customCoinstake_;
+    } else {
+        unsigned int nTxNewTime = 0;
+        if(!coinstakeCreator_->CreateProofOfStake(
+                block.nBits,
+                block.nTime,
+                nLastCoinStakeSearchTime,
+                txCoinStake,
+                nTxNewTime))
+            return false;
+
         block.nTime = nTxNewTime;
```

Comment:
Should be set to be equal to the coinstake tx timestamp.

Reply:
Not sure what you mean here, there's no such thing as "timestamp of a transaction".  The `nTxNewTime` (aka block time) is just what is used with hashing the coinstake.  On regtest and with custom coinstakes in the test, this does not matter, as each hash will satisfy the target (see other comment).